baybe/searchspace/continuous.py

```diff
@@ -235,20 +238,51 @@ def _drop_parameters(self, parameter_names: Collection[str]) -> SubspaceContinuo
 
     def transform(
         self,
-        data: pd.DataFrame,
+        df: pd.DataFrame | None = None,
+        /,
+        *,
+        allow_missing: bool = False,
+        allow_extra: bool | None = None,
+        data: pd.DataFrame | None = None,
     ) -> pd.DataFrame:
-        """See :func:`baybe.searchspace.discrete.SubspaceDiscrete.transform`.
+        """See :func:`baybe.searchspace.core.SearchSpace.transform`."""
+        # >>>>>>>>>> Deprecation
+        if not ((df is None) ^ (data is None)):
+            raise ValueError(
+                "Provide the dataframe to be transformed as argument to `df`."
+            )
 
-        Args:
-            data: The data that should be transformed.
+        if data is not None:
+            df = data
+            warnings.warn(
+                "Providing the dataframe via the `data` argument is deprecated and "
+                "will be removed in a future version. Please pass your dataframe "
+                "as positional argument instead.",
+                DeprecationWarning,
+            )
 
-        Returns:
-            The transformed data.
-        """
-        # Transform continuous parameters
-        comp_rep = data[[p.name for p in self.parameters]]
+        # Mypy does not infer from the above that `df` must be a dataframe here
+        assert isinstance(df, pd.DataFrame)
```

Comment:
uff, why not cast?

Reply:
Would have required several casts, and since it's part of the deprecation and will vanish anyway, went with the simplest solution

Reply:
i was thinking more of a `df = cast(pd.DataFrame, df)` or so in one line but it doesnt really matter

Reply:
Ok, then let's ignore 👍🏼